core/templates/dev/head/pages/exploration_editor/editor_tab/TrainingDataEditorPanelService.js

```diff
@@ -0,0 +1,165 @@
+                  $scope.trainingData.push({
+                    answer: answer,
+                    answerTemplate: answerTemplate
+                  });
+                });
+              };
+
+              $scope.init = function() {
+                _rebuildTrainingData();
+                $scope.newAnswerIsResolved = false;
+                $scope.answerSuccessfullyAdded = false;
+              };
+
+              $scope.removeAnswerFromTrainingData = function(answerIndex) {
+                answer = $scope.trainingData[answerIndex].answer;
+                TrainingDataService.removeAnswerFromAnswerGroupTrainingData(
+                  answer, answerGroupIndex);
+                $scope.trainingData.splice(answerIndex, 1);
+              };
+
+              $scope.exit = function() {
+                $uibModalInstance.close();
+              };
+
+              $scope.submitAnswer = function(newAnswer) {
+                $scope.newAnswerIsResolved = false;
+                $scope.answerSuccessfullyAdded = false;
+
+                var interactionId = stateInteractionIdService.savedMemento;
+
+                var rulesServiceName =
+                  AngularNameService.getNameOfInteractionRulesService(
+                    interactionId);
+
+                // Inject RulesService dynamically.
+                var rulesService = $injector.get(rulesServiceName);
+
+                var newAnswerTemplate = (
+                  ExplorationHtmlFormatterService.getAnswerHtml(
+                    newAnswer, stateInteractionIdService.savedMemento,
+                    stateCustomizationArgsService.savedMemento));
+
+                var classificationResult = (
+                  AnswerClassificationService.getMatchingClassificationResult(
+                    _explorationId, _stateName, _state,
+                    newAnswer, rulesService));
+                var newAnswerFeedbackHtml = 'Nothing';
+                var newAnswerOutcomeDest = classificationResult.outcome.dest;
+                if (classificationResult.outcome.hasNonemptyFeedback()) {
+                  newAnswerFeedbackHtml = (
+                    classificationResult.outcome.feedback.getHtml());
+                }
+                if (newAnswerOutcomeDest === _stateName) {
+                  dest = '<em>(try again)</em>';
+                }
+
+                $scope.newAnswerTemplate = newAnswerTemplate;
+                $scope.newAnswerFeedbackHtml = newAnswerFeedbackHtml;
+                $scope.newAnswerOutcomeDest = newAnswerOutcomeDest;
+
+                var classificationType = (
+                  classificationResult.classificationCategorization);
+
+                // If answer is explicitly classified then show the
+                // classification results to the creator.
+                if (classificationType === EXPLICIT_CLASSIFICATION ||
+                    classificationType === TRAINING_DATA_CLASSIFICATION) {
+                  $scope.newAnswerIsResolved = true;
```

Comment:
Maybe $scope.newAnswerIsAlreadyResolved (to indicate that this is already the case, rather than something that the creator has just done)?

Reply:
Done